### PR TITLE
Add Heartwood translation layer for file diffs

### PR DESCRIPTION
**Pull request by rudolfs**
To reduce the scope of #515.

### CODE REVIEW

File: src/lib/project.ts [hunk 1]
```diff
@@ -276,10 +276,42 @@ export class Project implements ProjectInfo {
...
+    const result = await new Request(
       `projects/${this.id}/commits/${commit}`,
       this.seed.addr,
     ).get();
+    result.stats["insertions"] = result.stats["additions"];
+    delete result.stats["additions"];
+    result.diff["added"] = result.diff["created"];
+    delete result.diff["created"];
+
+    for (const kind of ["added", "deleted", "modified"]) {
```
Comment (rudolfs) [inline src/lib/project.ts:288]: We'll add the missing type for `added` and `deleted` here. Was previously done like this: https://github.com/radicle-dev/radicle-interface/pull/515/files#diff-bf65359317875753c55e35ca05da15e82f77f5578edd8907a14aecdd206fdd99R106.

**Comment by rudolfs** [comment]
The visual diff failure is expected, renamed `created` to `added`.

File: tests/e2e/project/commit.spec.ts [hunk 2]
```diff
@@ -61,7 +63,7 @@ test("created file", async ({ page }) => {
     `${projectFixtureUrl}/remotes/${bobRemote}/commits/d6318f7f3d9c15b8ac6dd52267c53220d00f0982`,
   );
   await expect(
-    page.locator("text=1 file(s) created with 9 addition(s) and 0 deletion(s)"),
+    page.locator("text=1 file(s) added with 9 insertion(s) and 0 deletion(s)"),
```
Comment (xphoniex) [inline tests/e2e/project/commit.spec.ts:66]: why not keep it as "created"? you've kept the one in [`FileDiff.svelte`](https://github.com/radicle-dev/radicle-interface/pull/565/files#diff-86922c6cf991e9927a1a37a653efc6e9c1db5dfacb4a99a772798d30d4b0f3dbR122) as created.

This should make `visual` pass as well?

Comment (rudolfs) [reply tests/e2e/project/commit.spec.ts:66]: Ah, true. Will do.

Comment (rudolfs) [reply tests/e2e/project/commit.spec.ts:66]: Trying to match the changes in #515.

Comment (xphoniex) [reply tests/e2e/project/commit.spec.ts:66]: I see `visual` still failing, why is that?

Comment (rudolfs) [reply tests/e2e/project/commit.spec.ts:66]: Because the wording changed:

> The visual diff failure is expected, renamed created to added.

Comment (xphoniex) [reply tests/e2e/project/commit.spec.ts:66]: I'm aware of that. I thought you reverted it back in this [diff](https://github.com/radicle-dev/radicle-interface/compare/208c156f28fe482b197782d1b1c742a36e53f985..12c10e26dd47dc86d87564956e3d286b4688f02d)?

Comment (rudolfs) [reply tests/e2e/project/commit.spec.ts:66]: I changed my mind : ) and reverted it back. I had another look at #515 and wanted to match what we're doing there, so we have these changes in `master`.

Comment (rudolfs) [reply tests/e2e/project/commit.spec.ts:66]: Once it's merged, the snapshots will update and that specific test won't fail in #515 anymore.

File: src/views/projects/SourceBrowser/FileDiff.svelte [hunk 3]
```diff
@@ -118,7 +118,7 @@
   <header class="file-header" on:click={collapse}>
     <div class="actions">
       <p class="txt-regular">{file.path}</p>
-      {#if mode === "created"}
+      {#if mode === "added"}
```
Comment (xphoniex) [inline src/views/projects/SourceBrowser/FileDiff.svelte:121]: wanna change the badge to `added` here as well?

Comment (xphoniex) [reply src/views/projects/SourceBrowser/FileDiff.svelte:121]: (github's not very good at rendering this, check `Badge` on the next line)

Comment (rudolfs) [reply src/views/projects/SourceBrowser/FileDiff.svelte:121]: Here you go: 2f27d2abe8eb8d2f15e5ee53c3ee71faabbd4b8b.

**Review by xphoniex — APPROVED** [review]

**Comment by xphoniex** [comment]
Feel free to squash them, no need to be 3 different commits.

**Pull request merged**